conventions/services/recapitulatif.py

```diff
@@ -518,3 +518,21 @@ def convention_validate(request: HttpRequest, convention: Convention):
         "opened_comments": opened_comments,
         "ConventionType1and2Form": convention_type1_and_2_form,
     }
+
+
+def convention_denonciation_validate(request, convention_uuid):
+    convention = Convention.objects.get(uuid=convention_uuid)
+    parent = convention.parent
+    date_denonciation = convention.date_denonciation
+    parent.statut = ConventionStatut.DENONCEE.label
+    parent.date_denonciation = date_denonciation
+    parent.save()
+    for avenant in parent.avenants.all():
+        avenant.statut = ConventionStatut.DENONCEE.label
+        avenant.date_denonciation = date_denonciation
+        avenant.save()
```

Comment:
Si on n'utilise pas les [signaux](https://docs.djangoproject.com/fr/4.2/ref/signals/) on peut mettre à jour plus rapidement en utilisant un [`update()`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.update) 

```suggestion
    parent.avenants.all().update(statut=ConventionStatut.DENONCEE.label, date_denonciation = date_denonciation)
```